src/analysis/calculus/mean_value.lean

```diff
@@ -468,6 +488,57 @@ theorem convex.norm_image_sub_le_of_norm_fderiv_le {f : E → F} {C : ℝ} {s :
 hs.norm_image_sub_le_of_norm_has_fderiv_within_le
 (λ x hx, (hf x hx).has_fderiv_at.has_fderiv_within_at) bound xs ys
 
+/-- The mean value theorem on a convex set: if the derivative of a function is bounded by `C` on `s`,
+then the function is `C`-Lipschitz on `s`. Version with `fderiv` and `lipschitz_on_with`. -/
+theorem convex.lipschitz_on_with_of_norm_fderiv_le {f : E → F} {C : ℝ} {s : set E}
+  (hf : ∀ x ∈ s, differentiable_at ℝ f x) (bound : ∀x∈s, ∥fderiv ℝ f x∥ ≤ C)
+  (hs : convex s) : lipschitz_on_with (nnreal.of_real C) f s:=
+hs.lipschitz_on_with_of_norm_has_fderiv_within_le
+(λ x hx, (hf x hx).has_fderiv_at.has_fderiv_within_at) bound
+
+section converse_mean_value
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+          {E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+          {F' : Type*} [normed_group F'] [normed_space 𝕜 F']
+
+/-- Converse to the mean value inequality: if `f` is differentiable at `x₀` and `C`-lipschitz
+on a neighborhood of `x₀` then it its derivative at `x₀` has norm bounded by `C`. -/
+lemma has_fderiv_at.le_of_lip {f : E' → F'} {f' : E' →L[𝕜] F'} {x₀ : E'} (hf : has_fderiv_at f f' x₀)
+  {s : set E'} (hs : s ∈ 𝓝 x₀) {C : ℝ≥0} (hlip : lipschitz_on_with C f s) : ∥f'∥ ≤ C :=
```

Comment:
Can you do a version with the hypothesis `has_fderiv_within_at`?

Reply:
There should be a more general version, but it will be more work. The issue is you won't be able to use [op_norm_le_of_ball](https://leanprover-community.github.io/mathlib_docs/analysis/normed_space/operator_norm.html#continuous_linear_map.op_norm_le_of_ball) so the statement will probably include some `unique_diff` assumption. So I'd prefer to keep this for future work if it is ever needed.

Reply:
OK, sounds good!